snafu/config.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""Tools for setting up config arguments."""
+import os
+from typing import Any, Callable, Dict, Iterable, List, Mapping, Tuple, TypeVar, Union
+import argparse
+import configargparse
+
+
+_T = TypeVar("T")
+
+
+def check_file(file: str, perms: int = None) -> bool:
+    """
+    Check that the given file exists and has the given minimum permissions.
+
+    Essentially just a wrapper around ``os.access(os.path.abspath(file), perms)``.
+
+    Parameters
+    ----------
+    file : str
+        Path of file to check, can be relative or absolute.
+    perms : int, optional
+        Logical OR-ed Combination of ``os.R_OK``, ``os.W_OK`` and ``os.X_OK``. By default, will check
+        if file is readable. For instance, to check if a file is readable and writeable, set perms to
+        ``os.R_OK | os.W_OK``.
+
+    Returns
+    -------
+    bool
+        ``True`` if the given file exists and has the given perms, else ``False``
+    """
+
+    if perms is None:
+        perms = os.R_OK
+    perms |= os.F_OK
+    return os.access(os.path.abspath(file), perms)
```

Comment:
I want to suggest pathlib here, but I don't think this is ever going to get used on Windows.

Reply:
For sure, I would be surprised if we did use this on Windows. I'm not familiar with how this function would behave on a Window's system, nor with pathlib, but it seems like we might be OK for now. Pathlib seems to be a super nice way to do these operations, but because using os does work ok here, I'm going to resolve this for now.

Reply:
One thing we might want to think about, for future reference, is that ``os.access`` is pretty lazy when it comes to determining permissions, i.e., it doesn't check SELinux permissions: https://bugs.python.org/issue42481